src/azure-cli/azure/cli/command_modules/hdinsight/_validators.py

```diff
@@ -145,3 +144,26 @@ def validate_worker(self, cmd, namespace):  # pylint: disable=unused-argument
                         self.resource_name)
         except CLIError as e:
             self.exception = e
+
+
+def validate_timezone_name(namespace):
+    if namespace.timezone:
+        from .util import AUTOSCALE_TIMEZONES
+        zone = next((x for x in AUTOSCALE_TIMEZONES if x.lower() == namespace.timezone.lower()), None)
+        if not zone:
+            raise CLIError(
+                "Invalid time zone: '{}'. Run 'az hdinsight autoscale list-timezones' for values.".format(
+                    namespace.timezone))
+        namespace.timezone = zone
+        return zone
+    return None
+
+
+def validate_time(namespace):
+    if namespace.time:
+        hour, minute = namespace.time.split(':')
+        if len(hour) < 2:
+            hour = '0' + hour
+        if len(minute) < 2:
+            minute = minute + '0'
```

Comment:
>minute = minute + '0' [](start = 12, length = 21)

Why add zero in the end?

Reply:
If customer provide the 9:00, RP can accept the request but will fail in executing. 09:00 can solve this issue.
So the original logic is handling 9:00, 8:00.
I have refactored the code.

---
In reply to: [466341310](https://github.com/Azure/azure-cli/pull/14692#discussion_r466341310) [](ancestors = 466341310)